src/oc_erchef/schema/deploy/keys_update_trigger@users_email_functional_index.sql

```diff
@@ -0,0 +1,45 @@
+-- Deploy keys_update_trigger
+--
+-- This keeps the keys table consistent when the clients and user tables update their keys
+--
+BEGIN;
+
+CREATE OR REPLACE FUNCTION add_key() RETURNS TRIGGER AS $add_key$
+   BEGIN
+     IF NEW.public_key IS NOT NULL THEN
+       INSERT INTO keys
+         (id, key_name, public_key, key_version, created_at,  expires_at) VALUES
+         (NEW.id, 'default', NEW.public_key, NEW.pubkey_version, now(), 'infinity'::timestamp);
+     END IF;
+     RETURN NULL; -- result is ignored since this is an AFTER trigger
+   END;
+$add_key$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION update_key() RETURNS TRIGGER AS $update_key$
+   BEGIN
+     IF NEW.public_key IS NOT NULL THEN
+       UPDATE keys SET public_key = NEW.public_key, key_version = NEW.pubkey_version, expires_at = 'infinity'::timestamp
+         WHERE id = NEW.id AND key_name = 'default';
+       INSERT INTO  keys (id, key_name, public_key, key_version, created_at, expires_at)
+         SELECT NEW.id, 'default', NEW.public_key, NEW.pubkey_version, now(), 'infinity'::timestamp
+         WHERE NOT EXISTS (SELECT 1 FROM keys WHERE id = NEW.id AND key_name = 'default');
+     ELSE
```

Comment:
This is a port of the existing logic --  but it at always makes me say, "What what?" when I read it.  Could you add a comment explaining why a null key causes a deletion? (v0 api behavior compatibility)

Reply:
done!